bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -117,13 +123,19 @@ public static Object convertValues(Type targetType, Object inputValue) {
 
     public static Object castValues(Type targetType, Object inputValue) {
         switch (targetType.getTag()) {
-            case TypeTags.INT_TAG:
             case TypeTags.SIGNED32_INT_TAG:
+                return anyToSigned32(inputValue);
             case TypeTags.SIGNED16_INT_TAG:
+                return anyToSigned16(inputValue);
             case TypeTags.SIGNED8_INT_TAG:
+                return anyToSigned8(inputValue);
             case TypeTags.UNSIGNED32_INT_TAG:
+                return anyToUnsigned32(inputValue);
             case TypeTags.UNSIGNED16_INT_TAG:
```

Comment:
Can we cover these with few mote tests

Reply:
I pulled and pushed, now the patch coverage is 22%, Now also the above lines are not covered by tests are shown. However I set the breakpoints and debugged the tests. In `testCastingToIntSubtypesInUnionNegative` tests these lines are getting hit. So this may be a bug with the patch tool.